packages/material-ui/src/Input/Input.js

```diff
@@ -398,6 +479,7 @@ class Input extends React.Component {
       onFocus,
       onKeyDown,
       onKeyUp,
+      outlined,
```

Comment:
From context? (`variant`)

Reply:
It is, however I have applied it to context by reading props from the input, so that props such as this are applied to the input which seems to make the most sense, but are available in context

Reply:
`variant` is a TextField prop, so it perhaps makes more sense to pass that down to  FormControl, and add it to context there.

Reply:
Is that solution suitable for cases where consumers are not using `TextField`? In the current `TextField` implementation, `variant` will be passed into `InputComponent` and then grabbed from `context`. It's roundabout, but might that make sense for someone consuming `Input` and `FormControl`?

Reply:
Why is outline needed on FormControl? It doesn't seem to do anything